src/OpenTelemetry.AutoInstrumentation/DuckTyping/DuckType.cs

```diff
@@ -328,7 +328,11 @@ private static ModuleBuilder CreateTypeAndModuleBuilder(Type typeToDeriveFrom, T
             // If the proxy type definition is an interface we create an struct proxy
             // If the proxy type definition is an struct then we use that struct to copy the values from the target type
             parentType = typeof(ValueType);
-            typeAttributes = TypeAttributes.Public | TypeAttributes.AnsiClass | TypeAttributes.BeforeFieldInit | TypeAttributes.SequentialLayout | TypeAttributes.Sealed | TypeAttributes.Serializable;
+            typeAttributes = TypeAttributes.Public | TypeAttributes.AnsiClass | TypeAttributes.BeforeFieldInit | TypeAttributes.SequentialLayout | TypeAttributes.Sealed
+#pragma warning disable SYSLIB0050
+                             // TODO double check how to handle TypeAttributes.Serializable which is obsolete
```

Comment:
The usage of `TypeAttributes.Serializable` seems to have been used since its introduction into the Datadog automatic instrumentation. I've messaged my colleagues to understand if it's truly needed, and if it is then we'll need to suppress this compiler warning. Or perhaps if it's not needed for newer runtimes we can ifdef its usage, but I don't know if that's the case

Reply:
I have created #3799.

It can be done after merge, it was working fine on .NET8 with .NET6 binaries for a while. No rush to changes.